util/context.go

```diff
@@ -0,0 +1,54 @@
+package util
+
+import (
+	"context"
+
+	"github.com/grpc/grpc-go/metadata"
+
+	"github.com/infobloxopen/atlas-app-toolkit/auth"
+	"github.com/infobloxopen/atlas-app-toolkit/gateway"
+	"github.com/infobloxopen/atlas-app-toolkit/requestid"
+)
+
+// ForwardContext set to outgoing context request_id, auth_token and X-Forwarded-For header value
+func ForwardContext(ctx context.Context) context.Context {
+	keys := []string{
+		auth.AuthorizationHeader,
+		requestid.DefaultRequestIDKey,
+		gateway.XForwardedFor,
+	}
```

Comment:
Probably add tracing context header will be good as well, see https://github.com/census-instrumentation/opencensus-go/blob/v0.22.4/trace/trace.go#L113 for more information